src/Agent.Worker/ContainerOperationProvider.cs

```diff
@@ -973,29 +973,9 @@ private static string KeepAllowedLength(string name, int allowedLength, string s
 
         private static void ThrowIfAlreadyInContainer()
         {
-            if (PlatformUtil.RunningOnWindows)
+            if (PlatformUtil.DetectDockerContainer())
```

Comment:
This change seems slightly risker than the other changes in the PR.  I don't see any obvious patterns but we might want to split it out.  

[Not a blocker if you're confident about it + tested, otherwise might be good to supress and consider for another PR]

Reply:
That's fair... just from code inspection it's logically equivalent, though looks like in one error case a different exception would be thrown. Will bring back the old code.